src/VoiceProtocol.h

```diff
@@ -0,0 +1,47 @@
+// Copyright 2005-2020 The Mumble Developers. All rights reserved.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file at the root of the
+// Mumble source tree or at <https://www.mumble.info/LICENSE>.
+
+#ifndef MUMBLE_VOICEPROTOCOLTYPE_H
+#define MUMBLE_VOICEPROTOCOLTYPE_H
+
+#include "crypto/CipherType.h"
+#include "crypto/CryptStateFactory.h"
+
+#include <exception>
+#include <memory>
+#include <string>
+
+enum class VoiceTransportType { INVALID = 0, UDP = 1 };
+
+class VoiceProtocolException : public std::exception {};
+
+class VoiceProtocol {
+protected:
+	bool m_valid;
+	std::string m_protocolName;
+	explicit VoiceProtocol(const std::string &s);
+
+public:
+	VoiceTransportType m_transport;
+	virtual bool operator==(const VoiceProtocol &rhs) const;
+	virtual std::string toString() const;
+	static std::shared_ptr< VoiceProtocol > fromString(const std::string &s);
+	bool isValid() const;
+	VoiceProtocol();
+	virtual ~VoiceProtocol(){};
```

Comment:
Imo ordering the declarations in this way, allows a quicker overview of the class.
```suggestion
	static std::shared_ptr< VoiceProtocol > fromString(const std::string &s);
	
	VoiceProtocol();
	virtual ~VoiceProtocol() = default;

	virtual std::string toString() const;
	bool isValid() const;
	virtual bool operator==(const VoiceProtocol &rhs) const;
```